src/sentry/utils/arroyo_producer.py

```diff
@@ -60,3 +66,30 @@ def _shutdown(self) -> None:
 
         if self._producer:
             self._producer.close()
+
+
+def get_arroyo_producer(
```

Comment:
I like this approach but the rollout of this change would be too risky this way. If something goes wrong we would need a rollback in an incident istuaiton.

I'd recommend rolling out via feature flag.
- WE can have an option with the rollout state of each producer (false by default). This would be a dictionary. The producer is identified with an arbitrary string
- get_arroyo_producer checks the option, if the rollout for the specified producer is True it returns the producer, otherwise it returns None
- If the application code receives None it creates the producer the old way.
- When the producers are all rolled out we remove the old code.

Reply:
I would advise against such an aggressive rollout, but at least you will have a way to rollback Quickly so ok.